host/xtest/xtest_20000.c

```diff
@@ -350,42 +362,63 @@ static int obj_corrupt(TEEC_UUID *p_uuid, void *file_id,
 		}
 
 		fd = fopen(name, "r+");
-		if (!fd)
+		if (!fd) {
+			perror("Failed to open file");
+			tee_res = TEEC_ERROR_ACCESS_DENIED;
 			goto exit;
+		}
 
-		if (DUMPFILE)
-			dump_file(fd, buffer, sizeof(buffer));
+		dump_file(fd);
 
-		if (0 != fseek(fd, offset, SEEK_SET))
+		if (0 != fseek(fd, offset, SEEK_SET)) {
+			perror("Failed to seek file");
+			tee_res = TEEC_ERROR_BAD_PARAMETERS;
 			goto exit;
+		}
 
-		if (SIZE*NUMELEM != fread(&byte, SIZE, NUMELEM,fd))
+		res = fread(bytes, SIZE, NUMELEM, fd);
+		if (res != (int)(SIZE * NUMELEM)) {
```

Comment:
Did you try with `(SIZE * NUMELEM) != 1`? It seems the tests with `CORRUPT_FILE_LAST_BYTE` will always fail.